src/api/general-services/pipeline-status.js

```diff
@@ -0,0 +1,44 @@
+const AWS = require('../../utils/requireAWS');
+const ExperimentService = require('../route-services/experiment');
+const config = require('../../config');
+
+const getStepsFromExecutionHistory = () => [];
+
+const getPipelineStatus = async (experimentId) => {
+  const { executionArn } = await (new ExperimentService()).getPipelineHandle(experimentId);
+
+  const stepFunctions = new AWS.StepFunctions({
+    region: config.awsRegion,
+  });
+  const execution = await stepFunctions.describeExecution({
```

Comment:
how is this going to work in development (when clusterEnv = "development")? In development, we don't have step functions, but we have that dummy lambda.

Reply:
Luckily, Inframock does have step functions :-)
https://github.com/biomage-ltd/inframock/blob/master/docker-compose.yaml#L23

Reply:
hmm interesting, so what is the point of that lambda that I saw we use in `development` (for example, see lines 9-23 in `api/src/api/general-services/pipeline-manage/constructors/create-new-job-if-not-exist.js`)?

Reply:
> hmm interesting, so what is the point of that lambda that I saw we use in `development` (for example, see lines 9-23 in `api/src/api/general-services/pipeline-manage/constructors/create-new-job-if-not-exist.js`)?

In development they still run as step functions, it's just that the steps execute lambdas instead of launching jobs in k8s.

Reply:
The lambda is what is executed on entering the state if you have a [Task State](https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-task-state.html). The state is left when it completes. This is a sample reverse-ordered sequence of events (start at the bottom):

```
        {
            "timestamp": "2021-03-09T19:59:35.666000+01:00",
            "type": "TaskStateExited",
            "id": 6,
            "previousEventId": 5,
            "stateExitedEventDetails": {
                "name": "DeleteCompletedPipelineWorker"
            }
        },
        {
            "timestamp": "2021-03-09T19:59:35.665000+01:00",
            "type": "TaskSucceeded",
            "id": 5,
            "previousEventId": 4,
            "taskSucceededEventDetails": {
                "resourceType": "lambda",
                "resource": "invoke"
            }
        },
        {
            "timestamp": "2021-03-09T19:59:35.123000+01:00",
            "type": "TaskStarted",
            "id": 4,
            "previousEventId": 3,
            "taskStartedEventDetails": {
                "resourceType": "lambda",
                "resource": "invoke"
            }
        },
        {
            "timestamp": "2021-03-09T19:59:35.122000+01:00",
            "type": "TaskScheduled",
            "id": 3,
            "previousEventId": 2,
            "taskScheduledEventDetails": {
                "resourceType": "lambda",
                "resource": "invoke",
                "region": "eu-west-1",
                "parameters": "{\"FunctionName\":\"arn:aws:lambda:eu-west-1:000000000000:function:remove-previous-pipeline-containers\"}"
            }
        },
        {
            "timestamp": "2021-03-09T19:59:35.105000+01:00",
            "type": "TaskStateEntered",
            "id": 2,
            "previousEventId": 0,
            "stateEnteredEventDetails": {
                "name": "DeleteCompletedPipelineWorker"
            }
        },
```

Reply:
that makes sense now, thanks both for explaining!